File: solutions/go/src/playground/algorithm/invert_binary_tree_test.go
package algorithm

import (
	"testing"
	"github.com/stretchr/testify/assert"
)

func TestInvertTreeCase1(t *testing.T) {
	root := &TreeNode{Val: 4}
	root.Left = &TreeNode{Val: 2}
	root.Right = &TreeNode{Val: 7}
	root.Left.Left = &TreeNode{Val: 1}
	root.Left.Right = &TreeNode{Val: 3}
	root.Right.Left = &TreeNode{Val: 6}
	root.Right.Right = &TreeNode{Val: 9}

	inverted := invertTree(root)

	assert.Equal(t, root.Val, inverted.Val)
	assert.Equal(t, root.Right.Val, inverted.Left.Val)
	assert.Equal(t, root.Left.Val, inverted.Right.Val)
	assert.Equal(t, root.Right.Right.Val, inverted.Left.Left.Val)
	assert.Equal(t, root.Right.Left.Val, inverted.Left.Right.Val)
	assert.Equal(t, root.Left.Right.Val, inverted.Right.Left.Val)
	assert.Equal(t, root.Left.Left.Val, inverted.Right.Right.Val)
}

File: solutions/java/src/test/java/playground/algorithm/InvertBinaryTreeTest.java
package playground.algorithm;

import org.junit.jupiter.api.Test;
import playground.algorithm.common.TreeNode;

import static org.junit.jupiter.api.Assertions.*;

class InvertBinaryTreeTest {

    private InvertBinaryTree solution = new InvertBinaryTree();

    @Test
    void case_1() {
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(2);
        root.right = new TreeNode(7);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(3);
        root.right.left = new TreeNode(6);
        root.right.right = new TreeNode(9);

        solution.invertTree(root);

        assertEquals(4, root.val);
        assertEquals(7, root.left.val);
        assertEquals(2, root.right.val);
        assertEquals(9, root.left.left.val);
        assertEquals(6, root.left.right.val);
        assertEquals(3, root.right.left.val);
        assertEquals(1, root.right.right.val);
    }
}